### PR TITLE
Overhaul the new texture modal

### LINKED ISSUE #139: [Feature] Improve the texture creation modal
### Is your feature request related to a problem?

The texture creation modal is currently very cramped, ugly, and confusing.

### Describe the feature you'd like

To mitigate the sheer amount of fields needed for adding several textures at once, I suggest that the new texture modal instead fill the entire screen and have one tab for each texture. This way, you don't end up with a ton of vertical layers, plus with a wider layout uses and paths can be represented more cleanly instead of being simply indented (I like how the add-on form handles downloads).

I was thinking of implementing this feature similarly to the gallery modal, where instead of a new page there's a fullscreen modal (this should help reduce redirects and make it easier to implement over a full page).

### Screenshot(s)

image

I didn't include it here, but I figured the JSON converter could be a button on the top right or something simple, as it's not used super often and doesn't need to take half of your screen. I was also thinking of maybe making a JSON serializer for the modal and using browser localstorage to add the ability to store and retrieve texture drafts.

### Notes

Parts of this were suggested by Rolf at one point for an overhaul of the texture edit modals, but I feel like having smaller modals works better there since it makes it easier to edit small things (which is what the texture edit modals are mostly used for) and makes the request size smaller since you're only making a request to edit the specific use, path, or base texture json with only that data.

I also feel like this should also go alongside an overhaul of the API backend—it's consistently broken in some way or the other as seen in #138 and #115 and I think a new modal would nicely facilitate some code cleanup.

### PR DESCRIPTION
**Pull request by 3vorp**
Visual overhaul of the new texture modal to be less cluttered and more usable. Also added an export button so users can 
save texture drafts and have better debug logs.

Comparison:
Screen Shot 2024-09-15 at 1 05 07 PM

Screen Shot 2024-09-15 at 1 05 27 PM

Closes #139

### CODE REVIEW

File: pages/texture/new-texture-modal.vue [hunk 1]
```diff
@@ -350,65 +384,95 @@ export default {
...
+				});
+				this.textures = cleaned;
 			} catch (err) {
 				console.error(err);
 				this.$root.showSnackBar(err, "error");
 			}
 		},
+		copyData() {
+			const data = JSON.stringify(this.cleanedData, null, 2);
+			navigator.clipboard.writeText(data);
```
Comment (TheRolfFR) [inline pages/texture/new-texture-modal.vue:417]: Not sure it's compatible with Firefox, check on MDN

File: pages/texture/new-texture-modal.vue [hunk 2]
```diff
@@ -350,65 +384,95 @@ export default {
...
+						});
+						return use;
+					});
+					return tex;
+				});
+				this.textures = cleaned;
 			} catch (err) {
 				console.error(err);
 				this.$root.showSnackBar(err, "error");
 			}
```
Comment (TheRolfFR) [inline pages/texture/new-texture-modal.vue:413]: Why a try catch?

File: pages/texture/new-texture-modal.vue [hunk 3]
```diff
@@ -350,65 +384,95 @@ export default {
...
-					name: u.name,
-					edition: u.edition,
-					paths: u.paths.map((p) => ({
-						name: p.name,
-						versions: p.versions,
-						mcmeta: p.mcmeta || false,
-					})),
-				})),
-			}));
 			axios
```
Comment (TheRolfFR) [inline pages/texture/new-texture-modal.vue:428]: We're still using axios and not fetch?

File: resources/css/webapp.css [hunk 4]
```diff
@@ -1244,3 +1244,15 @@ body .texture-avatar {
...
+.v-timeline-item__inner-dot {
+	border-radius: 4px;
+}
+
+.texture-json-editor {
+	width: 50vw;
+	height: 80vh;
+	display: flex;
+	flex-direction: column;
+}
```
Comment (TheRolfFR) [inline resources/css/webapp.css:1258]: What about mobile?

**Review by TheRolfFR — CHANGES_REQUESTED** [review]

File: index.js [hunk 5]
```diff
@@ -62,19 +62,20 @@ window.colorToHex = (color) => {
 		if (colorArr.length > 1) colorArr[1] = colorArr[1].replace("-", "");
 		return colors[colorArr[0]][colorArr.length > 1 ? colorArr[1] : "base"];
 	} catch (error) {
-		return "currentcolor";
+		return "primary";
```
[the remark this answers is not in the document]
Comment (3vorp) [reply index.js:65]: this fixed the scrollbar on colored pages so I think this uses vuetify colors? not sure

File: pages/texture/new-texture-modal.vue (hunk 1 above)
Comment (3vorp) [reply pages/texture/new-texture-modal.vue:417]: checked and it's fine, also we're using the same API for gallery sharing so I'd be surprised if there were problems

File: pages/texture/new-texture-modal.vue (hunk 2 above)
Comment (3vorp) [reply pages/texture/new-texture-modal.vue:413]: handles json.stringify errors

File: pages/texture/new-texture-modal.vue (hunk 3 above)
Comment (3vorp) [reply pages/texture/new-texture-modal.vue:428]: yeah the entire webapp uses axios rn, might be something to look into changing but I don't think it's worth the effort at least for now

File: resources/css/webapp.css (hunk 4 above)
Comment (3vorp) [reply resources/css/webapp.css]: fixed with modal changes

**Pull request merged**